src/field4d.hpp

```diff
@@ -147,24 +147,37 @@ class Field4D
     Periodic_function<double>& vector(int i)
     {
         assert(i >= 0 && i <= 2);
+        if (components_[i+1] == nullptr) {
```

Comment:
Is this performance optimal? I just don't know. If the main code has something like
for (int ig = 0; ig < ng; ig++) {
  density.component(0).f_pw_loca(ig)
...
}
will the 'if` statement here be checked on each iteration of ig?

Reply:
suggestion to switch to assert()

Reply:
I don't know if the compiler would move component(0) outside the for loop, perhaps if density is const. I did add it because python in this case won't die because of the illegal memory access.
You are right, such for loops appear, I'll remove the if statement.